### PR TITLE
Finish refactor ls

### PR DESCRIPTION
**Pull request by RheingoldRiver**
I broke up dual_threshold_stats_convert a lot, the others are more straightforward

### CODE REVIEW

File: etl/pad/raw/skills/en_leader_skill_text.py [hunk 1]
```diff
@@ -119,6 +115,23 @@ def threshold_stats_text(self, intro, above, threshold, is_100):
             skill_text += threshold + '% HP'
         return skill_text
 
+    def dual_threshold_stats_part_full_hp_text(self, intro, above):
+        skill_text = intro
+        if above:
+            skill_text += ' when HP is full'
+        else:
+            skill_text += ' when HP is not full'
```
Comment (nachoapps) [inline etl/pad/raw/skills/en_leader_skill_text.py]: Simplify to:
skill_text += ' when HP {} full'.format('is' if above else 'is not')

File: etl/pad/raw/skills/en_leader_skill_text.py [hunk 2]
```diff
@@ -119,6 +115,23 @@ def threshold_stats_text(self, intro, above, threshold, is_100):
...
+        return skill_text
+
+    def dual_threshold_stats_part_threshold_text(self, intro, above, threshold):
+        skill_text = intro
+        if above:
+            skill_text += ' when above '
+        else:
+            skill_text += ' when below '
```
Comment (nachoapps) [inline etl/pad/raw/skills/en_leader_skill_text.py]: Similar.

File: etl/pad/raw/skills/en_leader_skill_text.py [hunk 3]
```diff
@@ -129,6 +142,40 @@ def combo_match_text(self, intro, min_combos, max_combos, up_to, max_mult):
...
         return intro + attr_text + max_attr_text
     
+    def multi_of_one_attribute_match_text(self, intro, min_match, attr_text, max_mult, max_match):
+        skill_text = intro
+        skill_text += ' when matching {}'.format(min_match)
+        if not max_mult:
+            skill_text += '+'
+        try:
+            skill_text += ' {} combos'.format(attr_text)
+        except Exception as e:
```
Comment (nachoapps) [inline etl/pad/raw/skills/en_leader_skill_text.py]: Why would this fail?

File: etl/pad/raw/skills/en_leader_skill_text.py [hunk 4]
```diff
@@ -119,6 +115,23 @@ def threshold_stats_text(self, intro, above, threshold, is_100):
             skill_text += threshold + '% HP'
         return skill_text
 
+    def dual_threshold_stats_part_full_hp_text(self, intro, above):
```
Comment (nachoapps) [inline etl/pad/raw/skills/en_leader_skill_text.py]: General note for all these items, you neglected to type all the params. If you don't have time to put them in I'll go back and do it later, but I'm trying to keep everything typed in general.

File: etl/pad/raw/skills/en_leader_skill_text.py [hunk 5]
```diff
@@ -129,6 +142,40 @@ def combo_match_text(self, intro, min_combos, max_combos, up_to, max_mult):
...
+        if max_mult:
+            skill_text += ' up to {}x when matching {}'.format(max_mult, all_colors)
+        return skill_text
+
+    def mass_match_text(self, intro, min_count, or_more, attr, max_count, max_mult):
+        skill_text = intro
+        skill_text += ' when matching {}'.format(min_count)
+        if or_more:
+            skill_text += ' or more connected'
+        skill_text += '{} orbs'.format(attr)
```
Comment (nachoapps) [inline etl/pad/raw/skills/en_leader_skill_text.py]: I'm confused why doesn't this need a starting space?

File: etl/pad/raw/skills/en_leader_skill_text.py [hunk 6]
```diff
@@ -177,23 +224,30 @@ def five_orb_one_enhance_text(self, mult):
...
+        return self.concat_list_and(skill_parts)
 
     def multi_play_text(self, mult):
         return '{} when in multiplayer mode'.format(mult)
 
+    def dual_passive_stat_text(self, bonus1, bonus2, both_atk):
+        skill_text = bonus1 + '; ' + bonus2
+        if not both_atk:
+            return skill_text
+        skill_text += '; {}x ATK for allies with both Att.'.format(both_atk)
```
Comment (nachoapps) [inline etl/pad/raw/skills/en_leader_skill_text.py]: This seems like a great candidate for your 'skill_parts' pattern with concat_list_semicolons.

File: etl/pad/raw/skills/en_leader_skill_text.py [hunk 7]
```diff
@@ -227,37 +281,20 @@ def l_match_text(self, mult_text, reduct_text, attr):
...
-            else:
-                skill_text += reduct_text[0].upper() + reduct_text[1:]
-
-        if unbind_amt != 0:
-            skill_text += ' and reduce' if skill_text else 'Reduce'
-            skill_text += ' awoken skill binds by {} turns'.format(unbind_amt)
+    def add_combo_att_text(self, mult, attr_condition_text, bonus_combo):
+        skill_parts = [
+            mult,
+            'increase combo by {}'.format(bonus_combo) + attr_condition_text
```
Comment (nachoapps) [inline etl/pad/raw/skills/en_leader_skill_text.py]: If you're using format() on a line it's kind of strange to also use + afterwards, just template two substitutions.

File: etl/pad/raw/skills/leader_skill_text.py [hunk 8]
```diff
@@ -87,35 +83,27 @@ def multi_attribute_match_convert(self, ls):
...
-                all_colors = '+'.join([self.ATTRIBUTES[a] for a in attributes])
-                skill_text += ' up to {}x when matching {}'.format(fmt_mult(max_mult), all_colors)
-
-        return skill_text
+            attr_text = self.ATTRIBUTES[attributes[0]]
+            max_mult = fmt_mult(max_mult) if len(attributes) != min_match else None
+            max_match = len(attributes)
+            return self.multi_of_one_attribute_match_text(intro, min_match, attr_text, max_mult, max_match)
+        min_colors = '+'.join([self.ATTRIBUTES[a] for a in attributes[:min_match]])
+        alt_colors = '+'.join([self.ATTRIBUTES[a] for a in attributes[1:min_match + 1]]) \
```
Comment (nachoapps) [inline etl/pad/raw/skills/leader_skill_text.py]: There's a helper you can use to simplify these:
https://github.com/nachoapps/dadguide-data/blob/master/etl/pad/raw/skills/skill_common.py#L62

File: etl/pad/raw/skills/leader_skill_text.py [hunk 9]
```diff
@@ -377,19 +362,20 @@ def l_match_text(self, mult_text, reduct_text, attr):
...
 
     def orb_heal_convert(self, ls):
         atk = ls.atk
-        mult = self.fmt_multiplier_text(1, atk, 1)
-        shield = ls.shield
-        reduct_text = self.fmt_reduct_text(shield) if shield != 0 else None
-        return self.orb_heal_text(atk, mult, shield, reduct_text, ls.unbind_amt, ls.heal_amt)
+        mult = self.fmt_multiplier_text(1, atk, 1) if atk != 1 and atk != 0 else None
+        reduct_text = self.fmt_reduct_text(ls.shield) if ls.shield != 0 else None
+        unbind_amt = ls.unbind_amt if ls.unbind_amt != 0 else None
```
Comment (nachoapps) [inline etl/pad/raw/skills/leader_skill_text.py]: In general you should never compare against 0. Use the non-zero-ness of values to pick branches.

File: etl/pad/raw/skills/en_leader_skill_text.py (hunk 4 above)
Comment (RheingoldRiver) [reply etl/pad/raw/skills/en_leader_skill_text.py]: not time, but wrist pain yeah i'll let you do this sorry

File: etl/pad/raw/skills/en_leader_skill_text.py (hunk 1 above)
Comment (RheingoldRiver) [reply etl/pad/raw/skills/en_leader_skill_text.py]: done

File: etl/pad/raw/skills/en_leader_skill_text.py (hunk 2 above)
Comment (RheingoldRiver) [reply etl/pad/raw/skills/en_leader_skill_text.py]: done

File: etl/pad/raw/skills/en_leader_skill_text.py (hunk 3 above)
Comment (RheingoldRiver) [reply etl/pad/raw/skills/en_leader_skill_text.py]: removed (was in original code)

File: etl/pad/raw/skills/en_leader_skill_text.py (hunk 7 above)
Comment (RheingoldRiver) [reply etl/pad/raw/skills/en_leader_skill_text.py]: done

File: etl/pad/raw/skills/leader_skill_text.py (hunk 9 above)
Comment (RheingoldRiver) [reply etl/pad/raw/skills/leader_skill_text.py]: done

File: etl/pad/raw/skills/en_leader_skill_text.py (hunk 6 above)
Comment (RheingoldRiver) [reply etl/pad/raw/skills/en_leader_skill_text.py]: done

File: etl/pad/raw/skills/en_leader_skill_text.py (hunk 5 above)
Comment (RheingoldRiver) [reply etl/pad/raw/skills/en_leader_skill_text.py]: added todo

**Pull request merged**